lib/portlayer/event/event_manager.go

```diff
@@ -15,38 +15,31 @@
 package event
 
 import (
-	"github.com/vmware/vic/pkg/vsphere/session"
+	"github.com/vmware/vic/lib/portlayer/event/collector"
+	"github.com/vmware/vic/lib/portlayer/event/events"
 )
 
+// EventManager will provide a basic event pub/sub implementation
 type EventManager interface {
 
-	// AddMonitoredObject will add the object for event listening
-	AddMonitoredObject(ref string) error
+	// RegisterCollector a collector with the manager
+	RegisterCollector(collector.Collector)
 
-	// RemoveMonitoredObject will remove the object from event listening
-	RemoveMonitoredObject(ref string)
+	// Collectors returns registered collectors
+	Collectors() map[string]collector.Collector
 
-	// Register will for event callbacks
-	Register(caller string, callback func(Event, *session.Session))
+	// Subscribe for event callbacks
+	Subscribe(event interface{}, caller string, callback func(events.Event))
```

Comment:
You can still use the type-as-topic and just do reflection once, at init time:

```go
func topicName(kind interface{}) string {
	t := reflect.TypeOf(kind).Elem()
	return fmt.Sprintf("%s.%s", t.PkgPath(), t.Name())
}

var topic = topicName((*types.VmPoweredOffEvent)(nil))
```

Could also have a helper like so:

```go
type EventTopic interface {
	Topic() string
}

type eventTopic string

func (t eventTopic) Topic() string {
	return string(t)
}

func NewEventTopic(kind interface{}) EventTopic {
	t := reflect.TypeOf(kind).Elem()
	return eventTopic(fmt.Sprintf("%s.%s", t.PkgPath(), t.Name()))
}
```

And embed EventTopic interface in the existing Events interface.

Using this `topicName` at runtime would be ok too, at least avoids the New() garbage.  And you can use string keys for the map(s) instead of interface{}.